servicetalk-client-api/src/main/java/io/servicetalk/client/api/ServiceDiscovererEvent.java

```diff
@@ -63,6 +65,15 @@ public interface ServiceDiscovererEvent<ResolvedAddress> {
      */
     Status status();
 
+    /**
+     * The raw meta-data associated with this ServiceDiscovererEvent.
+     * Note: the result will be an unmodifiable collection.
+     * @return the raw meta-data associated with this ServiceDiscovererEvent.
+     */
+    default Map<String, Object> metadata() {
+        return ServiceDiscovererMetadata.EMPTY_MAP;
```

Comment:
Any particular reason for not using `ContextMap` instead of a `Map` with a new custom `Key`?
I opened https://github.com/apple/servicetalk/pull/2906 in case we can proceed with `ContextMap`.

Reply:
I see two main reasons. The first being that ContextMap is mutable so we can't return a shared instance. That makes it so a method of the form 
```default ContextMap metadata() {
  return new DefaultContextMap();
  }
  ```
  Which seem great.

The second is that ContextMap is keyed on the `Key` _instance_, not it's name, so the produces and consumers of metadata must share the exact same key making a strong API linkage between the two.

Reply:
1. See my PR #2906 that adds `ContextMaps` with various implementations, including empty, singleton and unmodifiable.

2. That's correct, see `HttpContextKeys` as an example of how a single key as shared. We can have a similar "key holder" class in client-api.